examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ObserverModeBillingClient.kt

```diff
@@ -0,0 +1,61 @@
+package com.revenuecat.purchasetester
+
+import android.app.Activity
+import android.content.Context
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.BillingClientStateListener
+import com.android.billingclient.api.BillingFlowParams
+import com.android.billingclient.api.BillingResult
+import com.android.billingclient.api.ProductDetails
+import com.android.billingclient.api.Purchase
+import com.android.billingclient.api.PurchasesUpdatedListener
+import com.revenuecat.purchases.Purchases
+
+object ObserverModeBillingClient : PurchasesUpdatedListener, BillingClientStateListener {
+    private lateinit var billingClient: BillingClient
+
+    fun start(context: Context) {
+        billingClient = BillingClient.newBuilder(context)
+            .setListener(this)
+            .enablePendingPurchases()
+            .build()
+        billingClient.startConnection(this)
+    }
+
+    fun purchase(
+        activity: Activity,
+        productDetails: ProductDetails,
+        offerToken: String?,
```

Comment:
I think the offerToken isn't really optional anymore, right? even the base plans also have an offerToken if I'm not mistaken.

Reply:
That is correct for subscriptions! But this method also allows for purchasing of one time purchases which don't have an offer token so it needs to be nullable for that 🤷‍♂️